final release/format_data.py

```diff
@@ -1,200 +1,266 @@
-    def rescale_imgs(images,dim=dim):
-        size = images.shape[0]
-        imgs = np.zeros((size, dim, dim))
-
-        for i in range(size):
-            imgs[i, :, :] = resize(images[i, :, :], (dim, dim))
-
-        return np.expand_dims(imgs, axis=3)
-
-    
-    
+        test_x = asu_fr
+        test_y = asu_fr_labels
 
     #Rescaling Images
-    sys.stdout = open(os.devnull, "w") # don't show printed output
+    sys.stdout = open(os.devnull, "w")  # don't show printed output
     
     train_x = rescale_imgs(train_x, dim)
     test_x = rescale_imgs(test_x, dim)
 
-    sys.stdout = sys.__stdout__ # show printed output
+    sys.stdout = sys.__stdout__  # show printed output
 
     # Convert class vectors to binary class matrices (for Keras Dense)
-    if conf_matrix == True:
+    if conf_matrix:
         train_y = np_utils.to_categorical(train_y, 2)
         test_y = np_utils.to_categorical(test_y, 2)
-
-        
+    
     return train_x, train_y, test_x, test_y    
 
-
-import numpy as np
-from keras.preprocessing.image import ImageDataGenerator
-
-# Adding tiny Gaussian Noise to ensure there are no vanishing and exploding gradients
+# Adding 
 def add_noise(image):
-        image += (10e-10)*np.random.randn(image.shape[0], image.shape[1], 1)
+        """ Adds tiny random Gaussian Noise to each image. 
+        
+        Applies the function to each image to ensure there are no vanishing
+        and exploding gradients. Fixes an Error that occoured when sections
+        of an image had pixel intensity values of zero due to formating 
+        errors, resulting in intenstiy gradients of zero that broke the 
+        arctan2 function in our custom keras layer"""
+        
+        image += 10e-10 * np.random.randn(image.shape[0], image.shape[1], 1)
         return image
 
     
-def data_gen(rotation_range=180, zoom_range=0.2, shift_range = 0.0, flip=True):
+def data_gen(rotation_range=180, zoom_range=0.2, shift_range=0.0, flip=True):
+    """ Initializes data generator."""
     
     # Defining Data Generator
     datagen = ImageDataGenerator(
-                rotation_range = rotation_range,
-                zoom_range = zoom_range,
-                width_shift_range = shift_range,
-                height_shift_range = shift_range,
-                horizontal_flip = flip,
-                vertical_flip = flip,
-                fill_mode = "nearest",
+                rotation_range=rotation_range,
+                zoom_range=zoom_range,
+                width_shift_range=shift_range,
+                height_shift_range=shift_range,
+                horizontal_flip=flip,
+                vertical_flip=flip,
+                fill_mode ="nearest",
                 preprocessing_function=add_noise)
     
     return datagen
 
 
-def data_pregenerate(images, labels, datagen, batch_size, data_mult, seed):
+def data_pregenerate(images, labels, datagen, batch_size, nb_epoch, seed):
     samples = images.shape[0]
+    
```

Comment:
The docstring has to be the very first line of the function. Move this up.

Reply:
ok